### ISSUE TITLE
[feature] Python ConanApi().install_reference() method argument "build" takes only lists

### ISSUE THREAD

**Issue by FaridPF** [conan-io/conan#12177]
I'm trying to transfer the dependencies download process from simple bat/sh scripts for two platforms to a single Python script and I'm trying to play with Conan class from _conans.client.conan_api_. 

I was able to download and build my deps from _install.reference method_, but faced an issue that the _'build'_ argument is not that clear, and only after digging into the  _'conans/client/graph/build_mode.py'_ file I found out that the named parameter needs to be a list and not a string. So the argument to it should be passed as _build=["missing"]_ rather that a regular string, for example. 

Not sure why that is the case but, would be nice to have a type annotation or better yet docstring to the method, since not all of the arguments are that obvious.

**Comment by memsharded** [comment]
Hi @FaridPF 

I am sorry, but we are already in 2.0-beta.3. The code for 1.X is only in maintenance, for regression fixes and adding the necessary features and backports to enable a smooth migration from 1.X->2.0. So we will not be working in this, specially because the 1.X api is not public neither documented, it is not intended for public usage, and could break due to changes at any time.

The API in Conan 2.0 will be public and documented, so feedback on it will be very welcome. Thanks!

**Comment by FaridPF** [comment]
@memsharded Thank you for your quick response , James!  May be you would be abele to provide any timeframe when public API is going live?

**Comment by memsharded** [comment]
We expect to do a couple of betas more (one per month, more or less), before going General Availability, you can check the releases in https://docs.conan.io/en/latest/changelog.html.

 The API docs in 2.0 will be in https://docs.conan.io/en/2.0/reference/python_api.html. They will not be declared stable in 2.0, but later in 2.X, because they need more use and feedback before being stabilized.

**Comment by memsharded** [comment]
Final location for the 2.0 docs for Python API is https://docs.conan.io/2/reference/extensions/python_api.html
